<div style="font-size:20px; font-weight:bold;">
    Sensitivity Analysis of the bottom layer properties (id="1")
</div>

In [42]:
import numpy as np

# the most probable values
most_probable_values = {
    'Young’s modulus (E)': 1000,
    'Poisson’s ratio (ν)': 0.2,
    'Maximum swelling pressure (σ)': 8,
    'Permeability (k)': 8e-13,
    'Initial water saturation (S_{wi})': 0.12
}

# the ranges
ranges = {
    'Young’s modulus (E)': (500, 2500),
    'Poisson’s ratio (ν)': (0.16, 0.35),
    'Maximum swelling pressure (σ)': (3.2, 13),
    'Permeability (k)': (1e-14, 1e-12),
    'Initial water saturation (S_{wi})': (0.1, 0.5)
}

# standard deviations as 1/4th of the range
std_devs = {param: (max_val - min_val) / 4 for param, (min_val, max_val) in ranges.items()}

def generate_values(mean, std_dev, min_val, max_val, num_values=20):
    values = np.random.normal(loc=mean, scale=std_dev, size=num_values-2)
    values = np.clip(values, min_val, max_val)
    values = np.concatenate(([min_val], values, [max_val]))
    values = np.unique(values)
    return np.sort(values)

# Generate 15 values for each parameter
generated_values = {}
for param, mean in most_probable_values.items():
    std_dev = std_devs[param]
    min_val, max_val = ranges[param]
    values = generate_values(mean, std_dev, min_val, max_val, num_values=15)
    generated_values[param] = values

for param, values in generated_values.items():
    print(f'{param}: {values}')

Young’s modulus (E): [ 500.          855.90822721  925.14033729  962.36498101  971.74326854
  991.04513078  998.02150633 1138.76790714 1447.4962192  1684.1291265
 1819.58644547 1914.89196224 2500.        ]
Poisson’s ratio (ν): [0.16       0.17037318 0.17296998 0.18101793 0.1854984  0.19131768
 0.19722801 0.20394214 0.20415341 0.21744331 0.25080274 0.35      ]
Maximum swelling pressure (σ): [ 3.2         3.3842207   4.55334604  6.01711563  6.86943695  6.86944472
  7.033429    8.79398104  9.05730201 10.54002848 10.64136245 11.7626
 11.82859758 13.        ]
Permeability (k): [1.00000000e-14 5.05201405e-13 6.24068852e-13 6.29264103e-13
 7.18985808e-13 7.71719173e-13 7.76487239e-13 8.15103574e-13
 8.27529507e-13 8.47214494e-13 8.91045249e-13 9.26138374e-13
 9.27992598e-13 9.95384583e-13 1.00000000e-12]
Initial water saturation (S_{wi}): [0.1        0.12458961 0.13301828 0.15793052 0.17746799 0.17815248
 0.21548637 0.26045641 0.5       ]


\begin{array}{|l|c|c|c|c|}
\hline
\text{Parameter} & \text{Units} & \text{Minimum} & \text{Maximum} & \text{Sigmoidal Steps} \\
\hline
\text{Young’s modulus (E)} & \text{MPa} & 500 & 2500 & [500, 594, 766, 814, 946, 959, 1000, 1147, 1262, 1340, 1392, 1753, 1833, 1904, 2500] \\
\hline
\text{Poisson’s ratio} (\nu) & \text{–} & 0.16 & 0.35 & [0.16, 0.165, 0.172, 0.176, 0.197, 0.2, 0.21, 0.221, 0.223, 0.23, 0.24, 0.24, 0.35]\\
\hline
\text{Maximum swelling pressure} (\sigma) & \text{MPa} & 3.2 & 13 & [3.2, 4.19, 5.12, 5.18, 6.53, 7.42, 7.73, 8.12, 9.94, 10.21, 10.4, 11.05, 11.45, 12.64, 13] \\
\hline
\text{Permeability} & \text{mD} & 1e-14 & 1e-12 & [1e-14, 6.486e-13, 6.652e-13, 7.443e-13, 7.713e-13, 8e-13, 8.077e-13, 8.722e-13, 8.732e-13, 8.771e-13, 9.29e-13, 9.977e-13, 1e-12] \\
\hline
\text{Initial water saturation} (S_{wi}) & \text{–} & 0.1 & 0.5 & [0.1, 0.117, 0.131,  0.191, 0.212, 0.216, 0.251, 0.274, 0.5]\\
\hline
\end{array}

<div style="font-size:16px; font-weight:bold;">
    Output parameters for the ML analysis
</div>

Displacement (y_axis)

porosity_avg

Saturation_avg

